src/individual_contribution_detection.py

```diff
@@ -0,0 +1,212 @@
+    if not na or not nb:
+        return False
+    if na == nb:
+        return True
+
+    ta, tb = tokens(na), tokens(nb)
+    if not ta or not tb:
+        return False
+
+    # Match by last name + first name or first initial
+    if ta[-1] == tb[-1] and (ta[0] == tb[0] or ta[0][0] == tb[0][0]):
+        return True
+
+    # Match by any long token
+    return any(len(t) > 3 and t in tb for t in ta)
+
+def contributor_names_from_files(root: Path) -> List[str]:
+    
+    """
+    Extract contributor names from standard project files (CONTRIBUTORS, AUTHORS, README).
+    Deduplicate and preserve first-seen capitalization.
+    """
+    
+    seen = OrderedDict()
+    for f in find_contributor_files(root):
+        for n in extract_names_from_text(f):
+            key = normalize(n)
+            if key and key not in seen:
+                seen[key] = n.strip()
+    return list(seen.values())
+
+def files_to_owner_map(root: Path, extractor: FileMetadataExtractor) -> Dict[str, Optional[str]]:
+    
+    """
+    Return a mapping of relative file paths to file owners from metadata.
+    Skips .git and known contributor text files.
+    """
+    
+    ignore = {"CONTRIBUTORS", "AUTHORS", "README", "README.MD", "README.TXT"}
+    mapping: Dict[str, Optional[str]] = {}
+    for p in root.rglob("*"):
+        if not p.is_file() or ".git" in p.parts:
+            continue
+        if p.name.upper() in ignore:
+            continue
+        try:
+            owner = extractor.get_author(p)
+        except Exception:
+            owner = None
+        rel = str(p.relative_to(root)).replace("\\", "/")
+        mapping[rel] = owner if owner and owner not in ("Unknown", "Author Unknown", "") else None
+    return mapping
+
+def build_canonical(metadata_owners: Iterable[str], contribs: Iterable[str]) -> Tuple[Dict[str, str], Dict[str, str]]:
+    
+    """
+    Build unified name mappings to link contributor names and file metadata owners.
+
+    Returns:
+        (owner_to_canonical, contrib_to_canonical)
+        Each maps raw names to a single canonical display name.
+    """
+    
+    owner_to_canonical: Dict[str, str] = {}
+    contrib_to_canonical: Dict[str, str] = {}
+
+    owners = [o for o in dict.fromkeys(metadata_owners) if o]
+    contribs_list = [c for c in dict.fromkeys(contribs) if c]
+
+    matched_owners = set()
+    for c in contribs_list:
+        canon = None
+        for o in owners:
+            if name_matches(c, o):
+                canon = c.strip()
+                owner_to_canonical[o] = canon
+                matched_owners.add(o)
+                break
+        contrib_to_canonical[c] = canon or c.strip()
+
+    for o in owners:
+        if o not in owner_to_canonical:
+            owner_to_canonical[o] = o.strip()
+
+    return owner_to_canonical, contrib_to_canonical
+
+def detect_individual_contributions_local(
+    project_root: Path,
+    *,
+    extractor: Optional[FileMetadataExtractor] = None
+) -> Dict[str, Dict]:
```

Comment:
```suggestion
def detect_individual_contributions_local(
    project_root: Path,
    *,
    extractor: Optional[FileMetadataExtractor] = None,
    include_unattributed: bool = True,
) -> Dict[str, Dict]:
```
Please add the `include_unattributed:bool=True`  to your function to update the signature to accept that argument, which sets a default value.

Reply:
thank you for catching this error. I have made the changes and added include_unattributed:bool=True.